sysrepo support on QNX platform

Hi,

I'd like to port sysrepo over to QNX. Below I've summarized the issues in no particular order. Let me know if you want more details for any of these issues. Aside from issue 4, I can submit patches for the rest. If you do want patches, how would you want them divided?

1. Linking `librt` is not necessary on QNX, see `target_link_libraries(sysrepo rt)` in `CMakeLists.txt`.
2. Linking `libdl` is not necessary on QNX, see `target_link_libraries(sysrepo-plugind sysrepo dl)` in `CMakeLists.txt`.
3. Some C files which have functions such as sleep, strdup and others require the feature test macro `#define _XOPEN_SOURCE 500` to be compiled correctly, since the `-std=c99` flag is used. The `__QNXNTO__` can be used to recognize the qcc compiler specifically. The files are: `application_changes_example.c`, `notif_subscribe_example.c`, `oper_data_example.c`, `rpc_subscribe_example.c`, `oven.c`, `sysrepo-plugind.c`, `sysrepoctl.c` and `sysrepocfg.c`.
4. `strndup`, `vdprintf`, `asprintf` and `vasprintf` aren't provided in QNX. Would it be possible to provide a `compat.c` similar to this [commit](https://github.com/CESNET/libyang/commit/0c5ec1bf4a0c7c49afe4f8b32951b105130c8177) in libyang? The affected files outside of the immediate src directory are: all in `src/executables` and `application_changes.c`.
5.  `vfork` called in `sysrepocfg.c` is defined in `#include <process.h>` on QNX.
6. On Linux `time_t` is defined as a `long int`. On QNX, it's defined as a `unsigned int`. Other than a couple of warnings at compile time, do you think this will affect anything?

Hi Michal,

Pertaining to issue 6, I've copy-pasted the warnings below:
```sh
/home/molex/sysrepo_qnx/src/lyd_mods.c: In function 'sr_lydmods_update_replay_support':
/home/molex/sysrepo_qnx/src/lyd_mods.c:2737:22: warning: format '%ld' expects argument of type 'long int', but argument 3 has type 'time_t {aka unsigned int}' [-Wformat=]
         sprintf(buf, "%ld", from_ts);

[ 27%] Building C object CMakeFiles/srobj.dir/src/shm_sub.c.o
/home/molex/sysrepo_qnx/src/shm_sub.c: In function 'sr_shmsub_notif_listen_module_get_stop_time_in':
/home/molex/sysrepo_qnx/src/shm_sub.c:2833:37: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
     if ((next_stop_time - cur_time) < 0) {
                                     ^
```

Hi Michal,

I just realized one more issue. However it's more generic to 32-bit systems. In `edit_diff.c`, a warning is raised about using a uint64 as a pointer (which could lead to issues in a 32-bit system). Thoughts?

```sh
[ 15%] Building C object CMakeFiles/srobj.dir/src/edit_diff.c.o
/home/molex/sysrepo_qnx/src/edit_diff.c: In function 'sr_diff_find_oper':
/home/molex/sysrepo_qnx/src/edit_diff.c:2080:29: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
                 *conn_ptr = (void *)conn_attr->value.uint64;
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,9 +203,10 @@ include_directories("${PROJECT_SOURCE_DIR}/src")
 include_directories(${PROJECT_BINARY_DIR})
 
 # dependencies
-# librt (shm_open, shm_unlink, not required on OSX)
-if(NOT APPLE)
-    target_link_libraries(sysrepo rt)
+# librt (shm_open, shm_unlink, not required on OSX or QNX)
+find_library(LIBRT rt)
+if(LIBRT)
+    target_link_libraries(sysrepo ${LIBRT})
 endif()
 
 # atomic

